extensions/wikia/Search/classes/Services/AbstractSearchService.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Wikia\Search\Services;
+
+
+abstract class AbstractSearchService {
```

Comment:
I'm concerned about this pattern because it doesn't ensure that all necessary methods are overridden. Using an interface would ensure it and provide developers with ready list of required methods.